### PR TITLE
KEYCLOAK-12744 change logo for RH-SSO

### PR DESCRIPTION
**Pull request by edewit**
image

### CODE REVIEW

File: themes/src/main/resources-product/theme/rh-sso-preview/account/theme.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+parent=keycloak-preview
+import=common/rh-sso
```
Comment (vmuzikar) [inline themes/src/main/resources-product/theme/rh-sso-preview/account/theme.properties]: RH-SSO (not Keycloak) is failing with the following exception when it tries to load any theme. The (most probable) cause is that the `common/rh-sso` dir is in the `rh-sso` theme and not in `rh-sso-preview`.
```
16:38:53,350 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (default task-5) Uncaught server error: java.lang.NullPointerException
	at org.keycloak.theme.ExtendingThemeManager.loadTheme(ExtendingThemeManager.java:129)
	at org.keycloak.theme.ExtendingThemeManager.getTheme(ExtendingThemeManager.java:97)
	at org.keycloak.theme.DefaultThemeManager.getTheme(DefaultThemeManager.java:26)
	at org.keycloak.services.resources.admin.info.ServerInfoAdminResource.setThemes(ServerInfoAdminResource.java:183)
	at org.keycloak.services.resources.admin.info.ServerInfoAdminResource.getInfo(ServerInfoAdminResource.java:100)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:138)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:517)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:406)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$0(ResourceMethodInvoker.java:370)
	at org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:356)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:372)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:344)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:137)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:100)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:440)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:229)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:135)
	at org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:356)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:138)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:215)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:227)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:590)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at org.keycloak.services.filters.KeycloakSessionServletFilter.doFilter(KeycloakSessionServletFilter.java:91)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:132)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow.deployment.GlobalRequestControllerHandler.handleRequest(GlobalRequestControllerHandler.java:68)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:269)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:78)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:133)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:130)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:249)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:78)
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:99)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:376)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.lang.Thread.run(Thread.java:748)

```

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@edewit Thanks for the update! I've briefly checked your changes and found a few issues.

We need to add `rh-sso-preview` here:
https://github.com/keycloak/keycloak/blob/434ea0965c49624832d635e0d07758391ed53d6a/services/src/main/java/org/keycloak/services/resources/admin/info/ServerInfoAdminResource.java#L174-L176

Favicon needs to be updated too:
https://github.com/keycloak/keycloak/blob/4393fbcb5be2d9a6949ef979b4ef5d3d14c64114/themes/src/main/resources/theme/keycloak-preview/account/index.ftl#L59

**Review by ssilvert — COMMENTED** [review]
@vmuzikar If your testing checks out I think we can merge.  The code looks fine.

We do need to improve how the logo is loaded.  Customers need to be able to easily replace the logo.  But as we have it now, they must name their logo "logo.svg" and put it in the specified spot.  I'll open a JIRA about this and we can address it after Tech Preview.  Looks like we should be using pf-c-page__header-brand-link.

**Comment by ssilvert** [comment]
@edewit @abstractj The JIRA is wrong.  This is not related to KEYCLOAK-4575.  

We need to unlink the pull request from KEYCLOAK-4575 and link it to the correct JIRA.  I'm not sure how to do the "unlink" part.

**Comment by ssilvert** [comment]
The correct JIRA was KEYCLOAK-12746.  I've linked this PR to it.

**Comment by abstractj** [comment]
> @edewit @abstractj The JIRA is wrong. This is not related to KEYCLOAK-4575.
> 
> We need to unlink the pull request from KEYCLOAK-4575 and link it to the correct JIRA. I'm not sure how to do the "unlink" part.

I transitioned the PR to open again, regarding the unlinking part. I believe only the Jira admin can do something, because it's automatic.

**Comment by vmuzikar** [comment]
@abstractj @ssilvert Actually, the correct JIRA is KEYCLOAK-12744.
KEYCLOAK-12746 is a follow-up to sync with PF guys about rcue as Stian wanted. I've transitioned KEYCLOAK-12746 back to open.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@edewit I've checked the latest changes.

I'm still getting the same NPE with RH-SSO but this time for a different reason. I did some investigation and it seems like we're excluding `keycloak-preview` theme when building RH-SSO:
https://github.com/keycloak/keycloak/blob/76aa199feee70dddcb75ef635abcd1abccee114d/themes/pom.xml#L50-L59
I tried to remove that and it works.

Found one more issue. We need to productise this:
https://github.com/keycloak/keycloak/blob/3731e36ece3e7a6e7b9a4d6049bde10c0e417df0/themes/src/main/resources/theme/base/account/messages/messages_en.properties#L23

Also, we still need to address this:
> We need to add `rh-sso-preview` here:
> https://github.com/keycloak/keycloak/blob/434ea0965c49624832d635e0d07758391ed53d6a/services/src/main/java/org/keycloak/services/resources/admin/info/ServerInfoAdminResource.java#L174-L176

**Comment by ssilvert** [comment]
@vmuzikar  Erik is on PTO now.  Do you want to take this over and finish it up?

@abstractj FYI  ^^^

**Comment by vmuzikar** [comment]
@ssilvert I could but I've got currently a lot on my plate. Let's discuss it in today's meeting.

**Comment by ssilvert** [comment]
Closing this as I've opened https://github.com/keycloak/keycloak/pull/6703

@vmuzikar Your three issues are fixed.  The first one concerning the pom.xml was already fixed in Alex's PR.

**Pull request closed without merging**